st2common/st2common/services/sensor_watcher.py

```diff
@@ -95,9 +95,18 @@ def start(self):
             self.connection.release()
 
     def stop(self):
+        LOG.debug('Shutting down sensor watcher.')
         try:
             if self._updates_thread:
                 self._updates_thread = eventlet.kill(self._updates_thread)
+
+            if self.connection:
+                channel = self.connection.channel()
```

Comment:
Because it runs inside `stop` it means this code would still not run in case sensor is not shut down cleanly and this would mean we will still have dead queues left behind.

Another, probably better approach, would be to use a non-durable and auto deletable queue (http://docs.celeryproject.org/projects/kombu/en/1.0/reference/kombu.entity.html#kombu.entity.Queue.durable).